core/player/src/player.ts

```diff
@@ -87,12 +87,41 @@ export class Player {
     commit: COMMIT,
   };
 
-  public readonly logger = new TapableLogger();
-  public readonly constantsController = new ConstantsController();
+  public readonly logger: TapableLogger = new TapableLogger();
+  public readonly constantsController: ConstantsController =
+    new ConstantsController();
   private config: PlayerConfigOptions;
   private state: PlayerFlowState = NOT_STARTED_STATE;
 
-  public readonly hooks = {
+  public readonly hooks: {
```

Comment:
Should we make an actual type for this now? Rather than inline it on the property declaration?

Reply:
I could go either way. IMO having it inline just makes it easier to updated as needed but open to making it a type